patina_dxe_core/src/locks/mutex.rs

```diff
@@ -0,0 +1,200 @@
+//! Mutex: enforces mutual exclusion for a critical section while leaving
```

Comment:
What is your rationale for keeping this internal to dxe_core as opposed to having some standard mutex's like this available in the SDK?

Reply:
Yeah, agree that needing a BS instance to initialize the mutex is kind of a pain (you can see what I had to do in the perf component as in the "remove_atomics_sdk" draft PR:  https://github.com/OpenDevicePartnership/patina/pull/975/commits/dff613e5f272780df554181ef3d1384c64a2f81e).  

But I actuallt prefer the result. As a matter of architecture, I'm kind of opinionated about allowing components to affect global hw state (like interrupt control) without going through central APIs (i.e. TPL for interrupts or GCD for memory map) to allow for proper coordination. in any case, we can discuss more in the SDK PR (feel free to start the convo there now, if you want ;)).